benchmarks/src/main/java/io/grpc/benchmarks/driver/LoadClient.java

```diff
@@ -0,0 +1,533 @@
+
+  private final Control.ClientConfig config;
+  private final ExponentialDistribution distribution;
+  private volatile boolean shutdown;
+  private final int threadCount;
+
+  ManagedChannel[] channels;
+  BenchmarkServiceGrpc.BenchmarkServiceBlockingStub[] blockingStubs;
+  BenchmarkServiceGrpc.BenchmarkServiceStub[] asyncStubs;
+  Recorder recorder;
+  private ExecutorService fixedThreadPool;
+  private Messages.SimpleRequest simpleRequest;
+  private final OperatingSystemMXBean osBean;
+  private long lastMarkCpuTime;
+
+  LoadClient(Control.ClientConfig config) throws Exception {
+    log.log(Level.INFO, "Client Config \n" + config.toString());
+    this.config = config;
+    // Create the channels
+    channels = new ManagedChannelImpl[config.getClientChannels()];
+    for (int i = 0; i < config.getClientChannels(); i++) {
+      channels[i] =
+          Utils.newClientChannel(
+              Epoll.isAvailable() ?  Transport.NETTY_EPOLL : Transport.NETTY_NIO,
+              Utils.parseSocketAddress(config.getServerTargets(i % config.getServerTargetsCount())),
+              config.hasSecurityParams(),
+              config.hasSecurityParams() && config.getSecurityParams().getUseTestCa(),
+              config.hasSecurityParams()
+                  ? config.getSecurityParams().getServerHostOverride() :
+                  null,
+              true,
+              Utils.DEFAULT_FLOW_CONTROL_WINDOW,
+              false);
+    }
+
+    // Create a stub per channel
+    if (config.getClientType() == Control.ClientType.ASYNC_CLIENT) {
+      asyncStubs = new BenchmarkServiceGrpc.BenchmarkServiceStub[channels.length];
+      for (int i = 0; i < channels.length; i++) {
+        asyncStubs[i] = BenchmarkServiceGrpc.newStub(channels[i]);
+      }
+    } else {
+      blockingStubs = new BenchmarkServiceGrpc.BenchmarkServiceBlockingStub[channels.length];
+      for (int i = 0; i < channels.length; i++) {
+        blockingStubs[i] = BenchmarkServiceGrpc.newBlockingStub(channels[i]);
+      }
+    }
+
+    // Determine no of threads
+    if (config.getClientType() == Control.ClientType.SYNC_CLIENT) {
+      threadCount = config.getOutstandingRpcsPerChannel() * config.getClientChannels();
+    } else {
+      threadCount = config.getAsyncClientThreads() == 0
+          ? Runtime.getRuntime().availableProcessors()
+          : config.getAsyncClientThreads();
+    }
+    // Use a fixed sized pool of daemon threads.
+    fixedThreadPool = Executors.newFixedThreadPool(threadCount,
+        new DefaultThreadFactory("client-worker", true));
+
+    // Create the load distribution
+    if (config.getLoadParams().getClosedLoop() != null) {
+      distribution = null;
+    } else if (config.getLoadParams().getPoisson() != null) {
+      // Mean of exp distribution per thread is <no threads> / <offered load per second>
+      distribution = new ExponentialDistribution(
+          (double) threadCount / config.getLoadParams().getPoisson().getOfferedLoad());
```

Comment:
If the mean of the exponential distribution is

1 / offered load

for one thread then the mean for N threads is 

1 / offered load / N   (prorate the offered load to each thread)

which is

N / offered load

This works as each thread takes independent samples from the distribution